src/components/panel/TextRenderer.tsx

```diff
@@ -103,14 +115,16 @@ const TextRenderer: FC<Props> = memo(({ htmlString, onReady }) => {
 
 
   const onMouseEnterTarget = (e: Event) => {
-    const target = e.currentTarget as Element
-    const idsValue = getAnnotationIds(target)
-    if (!idsValue) return
+    const target = e.currentTarget as HTMLElement
 
-    const idArr = idsValue.split(',')
-    setHoveredAnnotations(idArr)
+    const targets = getTextTargets(flippedMatchedAnnotationsMapRef.current)
```

Comment:
This can be a heavy calculation, especially when it being reexcuted on every mouse enter. Is there a more efficient way to  get these?